base/abstractarray.jl

```diff
@@ -2260,7 +2260,7 @@ function hash(A::AbstractArray, h::UInt)
     # hashes will often subsequently be compared by equality -- and equality between arrays
     # works elementwise forwards and is short-circuiting. This means that a collision
     # between arrays that differ by elements at the beginning is cheaper than one where the
-    # difference is towards the end. Furthermore, blindly choosing log(N) entries from a
+    # difference is towards the end. Furthermore, naively choosing log(N) entries from a
```

Comment:
`arbitrarily`?

Reply:
Though as I'm re-reading I'm confused by what the mitigation for not getting a bunch of zeros from a sparse array is

Reply:
The problem with sparse arrays is that zeros tend to be clustered in blocks and along dimensions, so you don't want to do things like pick the first `log(N)` elements, or all the elements in some dimension or along some diagonal or something with regular structure like that. So this does a fibonacci selection. Peudo-random with some low-entropy seed would probably also work.

Reply:
Right, but aren't all those mitigations important in the dense case as well? For two `size`-`(10, 10)` arrays, you could have the majority of elements differ but you keep sampling every 11th element, and you just pick out some structured subset (like the diagonal).

Just asking in case there's an additional opportunity to clarify the comment.

Reply:
Sparse here is used a bit loosely. If the array is a `DenseArray` that just happens to have a sparsity pattern, the same concern applies. This function is applicable to all `AbstractArray`s.